syft/messaging/plan.py

```diff
@@ -363,190 +482,82 @@ def build(self, *args):
-
-        # for every pair of id
-        for i in range(len(from_ids)):
-            # for every message of the plan
-            for j, msg in enumerate(self.readable_plan):
-                # look for the old id and replace it with the new one
-                self.readable_plan[j] = Plan._replace_message_ids(
-                    obj=msg,
-                    change_id=from_ids[i],
-                    to_id=to_ids[i],
-                    from_worker=from_worker,
-                    to_worker=to_worker,
-                )
-
-        return self
-
-    def replace_worker_ids(self, from_worker_id: Union[str, int], to_worker_id: Union[str, int]):
+    def __setattr__(self, name, value):
+        """Add new tensors or parameter attributes to the state and register them
+        in the owner's registry
         """
-        Replace occurrences of from_worker_id by to_worker_id in the plan stored
-        Works also if those ids are encoded in bytes (for string).
-
-        Args:
-            from_worker_id: Id of the the replaced worker.
-            to_worker_id: Id of the new worker.
-        """
-
-        id_pairs = [(from_worker_id, to_worker_id)]
-        if type(from_worker_id) == str:
-            to_worker_id_encoded = (
-                to_worker_id.encode() if type(to_worker_id) == str else to_worker_id
-            )
-            id_pairs.append((from_worker_id.encode(), to_worker_id_encoded))
-
-        for id_pair in id_pairs:
-            self.readable_plan = Plan._replace_message_ids(
-                obj=self.readable_plan,
-                change_id=-1,
-                to_id=-1,
-                from_worker=id_pair[0],
-                to_worker=id_pair[1],
-            )
-
-    @staticmethod
-    def _replace_message_ids(obj, change_id, to_id, from_worker, to_worker):
-        _obj = list()
-
-        for item in obj:
-            if isinstance(item, int) and (item == change_id):
-                _obj.append(to_id)
-
-            elif isinstance(item, type(from_worker)) and (item == from_worker):
-                _obj.append(to_worker)
-
-            elif isinstance(item, (list, tuple)):
-                _obj.append(
-                    Plan._replace_message_ids(
-                        obj=item,
-                        change_id=change_id,
-                        to_id=to_id,
-                        from_worker=from_worker,
-                        to_worker=to_worker,
-                    )
-                )
-
-            else:
-                _obj.append(item)
-
-        return tuple(_obj)
-
-    def _execute_readable_plan(self, *args):
-        # TODO: for now only one value is returned from a plan
-        result_ids = [sy.ID_PROVIDER.pop()]
+        object.__setattr__(self, name, value)
 
-        plan_res = self.execute_plan(args, result_ids)
-
-        return plan_res
+        if isinstance(value, FrameworkTensor):
+            self.state.state_ids.append(value.id)
```

Comment:
Is it not needed to add `name` in `self.state.keys`?

Reply:
No, keys are only really used for func2plan for people to create state elements and getting them through a nice name, but you're right I will remove them to transform the dict to a simple list like in https://pytorch.org/tutorials/beginner/examples_autograd/two_layer_net_custom_function.html